flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -179,6 +182,10 @@ public void release() {
 	@Nullable
 	BufferAndBacklog pollBuffer() {
 		synchronized (buffers) {
+			if (isBlockedByCheckpoint) {
```

Comment:
why the blocked partition would be pulled buffer since it was unavailable.

Reply:
Because there are redundant notifications for local channel which means we may notify data available even when the downstream task in the poll loop. Maybe we fix it in another ticket if we think it's a problem.

Reply:
I see, it is for the case of local channel. I think we should supplement a test to cover this change if possible.

Reply:
Also it is better to give some descriptions here to explain this situation for better understanding.